src/Server/HTTPHandler.cpp

```diff
@@ -727,6 +732,16 @@ void HTTPHandler::processQuery(
         return false;
     };
 
+    if (params.has("role"))
+    {
+        auto role_name = params.get("role");
```

Comment:
Can the `role` parameter occur multiple times? I mean don't we want to support something like
```
&role=role1&role=role2&role=role3
```

Reply:
Ah yes, I forgot to add this as a test scenario, as this is the prime candidate for that. I checked it beforehand, we will get only the first one with the current implementation of the headers container. I will add the test.

Reply:
Ideally, we want to support multiple; however, the current use cases for this feature require only a single role. To speed things up, I decided against parsing multiple roles out of the query parameters, as it looks to me like we will need to add the corresponding method in the query params container. Maybe worth leaving it out of scope, so I can implement it as a follow-up?

Reply:
I have added support for multiple roles. I completely overlooked that we can have a simple iterator over this collection, pointing to the first occurrence of the `role` parameter.